### ISSUE TITLE
Landcover and landuse features cannot be distinguished at mid and low-mid zoom levels

### ISSUE THREAD

**Issue by jeisenbe** [gravitystorm/openstreetmap-carto#3936]
Related to #3935

### Expected behavior
- It should be possible to see when an area lacks landcover/landuse mapping, even at low-mid zoom levels
- Landcover distinctions that are rendered should be visually distinctive, even at mid zoom levels
- Landcover/landuse should be similar in visual prominence to water areas and waterways
- Landcover/landuse should be similar in prominence to roads and railways

### Actual behavior
- Landcover is strongly lightened by 0.4 (out of 1 in HSLA) at z12, and increasing to 0.5 at z11 and 0.6 at z10 and below
- Light landcover, like farmland, is not clearly visible at z10 and below because of the lightening
- Distinctions between similar landcover colors (like orchard/wood, grass/farmland, scrub/heath) are not clearly visible at z12, and are not visible at all at z10 and below
- Due to issues with how mapnik renders small polygons, the fading increases at z9 and below as smaller areas of landcover are no longer rendered
- Since the fading changes with each zoom level, it's not really possible to recognize each type of landcover without extensive research at z10 to z12 especially.
- Roads, railroads, and waterways are much more prominent than landcover at z12 and below, especially at z10 and below
- Water areas are not faded, and therefore increase in prominence relative to landcover 

### Links and screenshots illustrating the problem
See screenshots in #3935

**Comment by jeisenbe** [comment]
Options to solve this issue (and #3935)

1) simplifying the landcover classes to a few types. 
- We already do this with built-up urban areas from z12. 
- PR #3670 would use this solution

2) Remove the fading and use the same landcover colors at all zoom levels
- The french style is similar to this option
- This would provide the best mapper feedback
- However, for roads and rivers to be sufficiently visible on rivers, it might be necessary to darken the low-zoom roads colors and darken rivers further.

3) Since that is a large change, we can also consider an intermediate step: reducing the fading and only using one level. 
- This would improve but not totally resolve this issue.
- I think it makes sense to merge a PR with this change first, and then consider if it's better to do option 1 (like PR #3670 - lighter woodland color so that roads and waterways are still fairly visible) or option 2 (no fading, darker roads/rivers at <= z11 - makes roads less prominent)

**Comment by jeisenbe** [comment]
Here's examples with the lightening reduced to 0.2 (half of the current lightening at z12) and only applied from z11. 

At z12 the main roads and rivers are wider and roads still have casing, so there's no problem with using the full landcover colors. This also provides one zoom level without the patterns. At this zoom level the built-up area is in #dddddd - that's what I found is the current color used at this zoom level. 

At z11 and below roads are narrow and no longer have the darker casing, so the 0.2 lightening still makes it possible to see the minor roads when they pass through woodlands.

### Luxembourg
z9 reduced fading
![z9-luxembourg-reduce-fade](https://user-images.githubusercontent.com/42757252/66999423-f1340600-f110-11e9-84fd-4ff1413fc63e.png)

z10 reduced fading
![z10-luxembourg-reduce-fade](https://user-images.githubusercontent.com/42757252/66999573-3e17dc80-f111-11e9-9d93-f40a1943d22a.png)

z11 reduced fading
![z11-luxembourg-reduce-fade](https://user-images.githubusercontent.com/42757252/66999488-132d8880-f111-11e9-8e1a-34986d75e3f9.png)

z12 reduced fading
![z12-luxembourg-reduce-fade](https://user-images.githubusercontent.com/42757252/66999584-42dc9080-f111-11e9-97a7-c354567712b4.png)

**Comment by jeisenbe** [comment]
## No fading - Luxembourg

Here's luxembourg without any landcover fading, but no other changes. This makes the rivers and minor roads a bit hard to follow in the woods (this is with the current river color; the proposed darker rivers help with this)

z9 no fade
![z9-luxembourg-nofade](https://user-images.githubusercontent.com/42757252/67005021-90aac600-f11c-11e9-9292-bcd61bf491d4.png)

z10 no fade
![z10-luxembourg-nofade](https://user-images.githubusercontent.com/42757252/67005037-a1f3d280-f11c-11e9-99b2-814f98a2bf71.png)

z11 no fade
![z11-luxembourg-nofade](https://user-images.githubusercontent.com/42757252/67005096-c485eb80-f11c-11e9-8f1c-e150ca1f95f3.png)

**Comment by jeisenbe** [comment]
## Option 2. - Luxembourg - darker rivers and roads, no landcover fading

This is an example of option 2: full landcover colors, and use slightly darker/more saturated roads and rivers at low zoom. This would resolve #3936 but might not resolve #3935

z9 Luxembourg no fade, darker roads and rivers
![z9-luxembourg-nofade-darkroads](https://user-images.githubusercontent.com/42757252/67006268-ac639b80-f11f-11e9-9e4f-e7d9b28c8e2b.png)

z10 no fade, dark roads
![z10-luxembourg-nofade-dark-roads](https://user-images.githubusercontent.com/42757252/67005913-b5a03880-f11e-11e9-8b04-f5d841a5f6ae.png)

z11 no fade, dark roads
![z11-luxembourg-nofade-dark-roads](https://user-images.githubusercontent.com/42757252/67005988-e6806d80-f11e-11e9-8a81-b2eaf024cbf5.png)

These are the diffs:
**road-colors.yaml**
```
low-zoom:
-      lightness: [62, 92]
-      chroma: [50, 40]
+      lightness: [55, 90]
+      chroma: [55, 45]
```
- Low zoom roads chroma is increased by 5 (in Lch) and lightness is reduced to a range of 55 to 90.

**road-colors-generated.mss**
```
@motorway-low-zoom: #d85576;
@trunk-low-zoom: #e98365;
@primary-low-zoom: #e7b46a;
@secondary-low-zoom: #dbe28c;
```

**roads.mss**
```
+@low-zoom-road: #aeaeae;
```
`#aeaeae` (Lch 71,0,0) is used at z12 for unclassified/residential and at z11 and below for tertiary and secondary, instead of the current `#bbb` (Lch 76,0,0)

**Comment by kocio-pl** [comment]
Reduced land areas fading looks OK for me at all the presented examples and it probably gives better results than current fading.

No fading creates limited visual mess at both z11 examples and nasty "military camouflage" effect on z9/z10. I mentioned it lately in different ticket as a forest problem, but lines also have their share with competing patterns here.

**Comment by jeisenbe** [comment]
> it probably gives better results than the current fading

OK, we can try that (Option 3, intermediate step). I think this will make it easier to consider making larger visual changes later.

Here are some more examples of the reduced landcover lightening, in southern Wales :

z12 Newport, Wales
![z12-newport-nofade-rivercolor](https://user-images.githubusercontent.com/42757252/67133197-bd5afc80-f246-11e9-9c50-17a3b83f8f90.png)

z11
![z11-newport-leastfade](https://user-images.githubusercontent.com/42757252/67133400-86d1b180-f247-11e9-969b-29576d55807f.png)

z10
![z10-newport-leastfade](https://user-images.githubusercontent.com/42757252/67137627-6c153200-f273-11e9-9adb-57f9359b45c2.png)

z9
![z9-cardiff-leastfade](https://user-images.githubusercontent.com/42757252/67137629-6e778c00-f273-11e9-935a-2091c8196f79.png)

z8
![z8-cardiff-leastfade](https://user-images.githubusercontent.com/42757252/67137631-73d4d680-f273-11e9-8d7c-f7a692ca9655.png)

z7
![z7-cardiff-leastfade](https://user-images.githubusercontent.com/42757252/67137633-77685d80-f273-11e9-95ea-7d111fcedec6.png)

**Comment by kocio-pl** [comment]
I think it still works OK. Even with dark green areas of heath, which make the background really solid, I cannot complain with readability of all the samples. It is still possible to focus on details when needed (like names, lines or peaks) without being distracted with dark patches of the landcover.

**Comment by jeisenbe** [comment]
### Option 2. - Wales - darker rivers and roads, no landcover fading

This shows the full-color landcover, plus slightly darker/higher chroma roads and waterways at low zoom, in Wales. We were discussing this in #3647.

z8 Wales, no fade, darker roads and rivers
![z8-cardiff-nofade-darker-water-river](https://user-images.githubusercontent.com/42757252/67258385-5601ae80-f4cb-11e9-9c96-ff7865325a1e.png)

z9 
![z9-cardiff-nofade-darker-water-river](https://user-images.githubusercontent.com/42757252/67258392-60bc4380-f4cb-11e9-9d2f-1bf18cabeb78.png)

z10
![z10-newport-nofade-darker-water-river](https://user-images.githubusercontent.com/42757252/67258396-687be800-f4cb-11e9-8eb3-913a42734ac0.png)

z11
![z11-newport-nofade-darker-water-river](https://user-images.githubusercontent.com/42757252/67258400-6ca80580-f4cb-11e9-8033-240d2b1ec682.png)

**Comment by kocio-pl** [comment]
I don't see that helps - and I would be really surprised if that could help, because dark forest spots are still there and this is the main part of a "camouflage effect" (heath doesn't make so many small spots, but it's also too strong).

**Comment by imagico** [comment]
@jeisenbe - Keep in mind that the current road colors were designed so that the color progression at all points and in all variants used fits into color space.  If you make these colors stronger this will not be the case any more - you will either have to abandon the principles of the current color progression or you will have to accept the clipping and the resulting differences and derivations from the original design principle.

In other words:  Road colors can be designed either for optimized human perception or optimized for the technical constraints of color reproduction on computer displays.

@kocio-pl - Your idea of a "camouflage effect" does not make much sense to me.  The idea of camouflage is to artificially create patterns in the same colors as something you want to disguise or to paint something you want to disguise in patterns and colors of something abundant naturally around it.  This requires an identity in patterns and colors between two different things.  This is exactly what we in color design have been trying hard to avoid.  If at all this is something that has been introduced by the color fading at lower zoom levels because the fading function messes with the color relationships and thereby actually introduces a true camouflage effect.  This is not something that would be emphasized by removing the fading.

That there are too many different landcover colors at the low zoom levels is a different matter - but there are possible solutions to this which have been discussed elsewhere.

**Comment by kocio-pl** [comment]
It is not as much the idea as just trying to name the immediate visual impression strong elements gave to me. It might be a side effect (and in fact I think this is not intended in any way), but the result is the lack of balance between areas, lines and nodes - areas (especially dark ones) eclipse all the rest. Fading them removes this problem.

> That there are too many different landcover colors at the low zoom levels is a different matter

I have clearly identified the problem with just one element - strong forest spots.

**Comment by imagico** [comment]
> I have clearly identified the problem with just one element - strong forest spots.

Is it possible that the problem you see is just a mismatch between the visual importance woods/forests have in the map and the mental importance you attribute to them in your mind and you therefore expect to be matched by the map?

It is after all not that the current wood/forest color in this style is very strong in comparison to other map styles including OSM-Carto variants - @jeisenbe pointed this out already.

**Comment by kocio-pl** [comment]
This is not a problem with forest color being too strong. On the high zoom levels it just does not create spots, so I never complained about it. The problem are only _strong spots_ (both things taken together), and they start appearing on the midzoom.

It also does not have anything with the meaning of these spots. It is purely visual sensation.

**Comment by imagico** [comment]
I don't think you have really understood my question.  Map viewing physiology cannot be regarded independent of the semantics of the data and the scale of the map.  Map viewing always connects what is being shown with the mental image the viewer has of the geography shown.  And if you see a 2cm forest patch on an 1:2M map that naturally connects to a very different mental image than if you see a 2cm forest patch on an 1:100k map.

You can try to describe the effect of a certain visual appearance independent of the semantics but neither you nor any other map viewer will actually perceive it this way.

**Comment by kocio-pl** [comment]
It is possible. Please rephrase it to make it simpler.

**Comment by jeisenbe** [comment]
I think what @kocio-pl was trying to say is that he thinks some of the rendering examples in Europe look like fashion "camo", e.g:

![camo](https://user-images.githubusercontent.com/42757252/67444386-c6cbd680-f643-11e9-842b-303618f04881.jpg)

This type of "camo" is ineffective at hiding the user, because it has hard edges and high contrast between light and dark patterns, but it seems to be popular for some articles of clothing.

But this effect is not limited to low zoom levels. Areas that are well-mapped can have all sorts of different colors in interesting patterns, if there are semi-natural areas mixed with meadow and farmland. Eg:

z13 Flamengos, Azores
![z13-flamengos-patterns](https://user-images.githubusercontent.com/42757252/67444497-4d80b380-f644-11e9-9021-425b51ff2e13.png)

z15
![z15-flamengos-before](https://user-images.githubusercontent.com/42757252/67444502-54a7c180-f644-11e9-8456-f6302730fce6.png)

z16 Anse de Rondinara
![z16-anse-de-rondinara-before](https://user-images.githubusercontent.com/42757252/67444517-68532800-f644-11e9-833f-a303980e75c5.png)

So it's important to be clear about what the problem is. My belief is that @kocio-pl feels that the dark forests make it too hard to follow linear features like roads, railways and perhaps waterways at mid-low zoom (z8 to z11), where these features are much narrower than in the examples above. Is that correct?

That should be discussed in #3935 - where I claim the current rendering creates the opposite problem: roads are too prominent.

**Comment by jeisenbe** [comment]
Re: low-zoom road colors

> Keep in mind that the current road colors were designed so that the color progression at all points and in all variants used fits into color space. If you make these colors stronger this will not be the case any more - you will either have to abandon the principles of the current color progression or you will have to accept the clipping and the resulting differences and derivations from the original design principle.

The change I made was only to the `low-zoom` road fill colors, which were changed from lightness: [62, 92] to [55, 90] and chroma: [50, 40] to [55, 45]. While this is certainly a change from the current script values, it is still more lightness and less chroma than `casing` or `low-zoom-casing`. The difference is that the low-zoom-fill would now be slightly closer to `low-zoom-casing` (which is only used on z12) and slight farther from `fill`, so there would still be a progression between each level, and there is no clipping. 

I don't think this would abandon the principles of the current color progression

Before:
```
    fill:
      lightness: [70, 97]
      chroma: [35, 29]
    casing:
      lightness: [50, 50]
      chroma: [70, 55]
    low-zoom:
      lightness: [62, 92]
      chroma: [50, 40]
    low-zoom-casing:
      lightness: [50, 70]
      chroma: [50, 65]
```

After
```
```
    fill:
      lightness: [70, 97]
      chroma: [35, 29]
    casing:
      lightness: [50, 50]
      chroma: [70, 55]
    low-zoom:
      lightness: [55, 90]
      chroma: [55, 45]
    low-zoom-casing:
      lightness: [50, 70]
      chroma: [50, 65]
```

**Comment by imagico** [comment]
This is still the case right now with the weaker fading, in particular also the difference between farmland and no mapped landcover is not visible in a lot of places - which is bad for mapper feedback. #3670 would address that.

**Comment by imagico** [comment]
As everyone can see i have always been against the fading - but i also see the motive for it, that the rendering of landcover at low zoom levels does not work very well.  I just disagreed with the approach to hide the problem rather than to address it.

What can be done to improve landcover depiction at the low zoom levels without data pre-processing (see #4901) is:

* aggregating landcovers to a smaller number of classes (#3670).
* Combining geometries within the distinguished visual classes with `ST_Collect()`.  This would reduce the AGG rendering bias resulting from rendering multiple polygons partially covering the same pixel (explained [here](https://imagico.de/map/polygon_rendering.php) in more detail) but would need to be thoroughly tested for performance implications.  And it would be of very limited effect without aggregating landcover classes.

Personally i would have no objections to removing the fading first and then considering the other possible improvements.  But my impression is that the difficulty on achieving consensus on that in the past has the same reason as the difficulty in agreeing on #3670 - the lack of agreement on the strategic direction and the widespread desire to adjust the low zoom appearance to be closer to commercial maps (like Google, Mapbox, Maptiler etc.).

It is possible though that with the OSMF increasingly embracing the postmodern commercial cartographic style of Google & co. support for OSM-Carto being a counter-weight and going in a distinctly different direction at the low zoom levels might be better.

Bottom line: I would encourage a new initiative to remove the fading, but it should be combined with a discussion on the strategy for the low zoom levels with the points above being possible components of such strategy.  Like with #4803/#5123 such change should come with a clear strategic outlook on where we want it to lead, even if implementing that outlook is not within reach at this time due to economic constraints.

**Comment by dch0ph** [comment]
> * This would reduce the AGG rendering bias resulting from rendering multiple polygons partially covering the same pixel

My impression is that this is only really significant for low zoom (Z9 and below), rather than Z10 and Z11? In a rural context, fields, woodlands etc. are covering multiple pixels, and in urban areas, where there are potentially multiple small landuses, such as car parks, pitches etc., sub-pixel landuses are filtered out. 

From my reading of #3935 and this issue, the only concern I could see about "no fading at Z10/Z11" was that the primary roads were more difficult to see against forest areas. This should be fixable by tweaking the road halo.

Inland water may be less visible with against a non-faded background, but the obvious next step is to use a higher chroma blue for inland water compared to ocean, #3895. In terms of strategic direction, I think these improvements at "low mid zoom" are only practical after dropping the low zoom fading, which monkeys with all the colours.
